url587829ef81_champdavis_KhovanovHomology_tgz-pJ8-KhovanovMapJ8/src/Knot.java

```diff
@@ -0,0 +1,227 @@
+import org.apache.commons.lang3.StringUtils;
+import java.util.*;
+
+public class Knot{
+    private Crossing[] crossings;
+    private Vertex[] vertices;
+    private String binarycode;
+    private Knot leftknot;
+    private Knot rightknot;
+    private int numcircles;
+    private ArrayList<SeifertCircle> seifertCircles;
+    private Smoothing smoothing;
+    
+   
+    public Knot(Crossing[] c){ //initial Knot
+        this.crossings = c;   
+        this.binarycode="";
+        this.seifertCircles = new ArrayList<SeifertCircle>();
+        
+        //initialize vertices array
+        int max=1;
+        int crossingmax;
+        for(int i=0; i<this.crossings.length; i++){  //max will tell us how many vertices there are
+            crossingmax = this.crossings[i].getMaxVertex();
+            if(max < crossingmax){ 
+                max = crossingmax;
+            }
+        }
+        
+        this.vertices = new Vertex[max];
+        //System.out.println(max);
+        for(int j=0; j<max; j++){//make the vertices
+            this.vertices[j] = new Vertex(j+1);
+        }
+        
+        for(int j=0; j<max; j++){//fill their pointers
+            this.vertices[j].setPointers(this.vertices[(j-1+max)%(max)], this.vertices[(j+1+max)%(max)]); //mod to wrap around   
+        }
+        
+        makeCircles();
+
+    }
+    
+    public Knot(Crossing[] c, String code, Smoothing l, Vertex[] v){ //sc = seifertcircles
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 65-114, lines 118-131 into separate methods.